src/httpserver.cpp

```diff
@@ -639,6 +641,38 @@ HTTPRequest::RequestMethod HTTPRequest::GetRequestMethod() const
     }
 }
 
+std::optional<std::string> HTTPRequest::GetQueryParameter(const std::string& key) const
+{
+    const char* uri = evhttp_request_get_uri(req);
+
+    return GetQueryParameterFromUri(uri, key);
+}
+
+std::optional<std::string> GetQueryParameterFromUri(const char* uri, const std::string& key)
+{
+    evhttp_uri* uri_parsed = evhttp_uri_parse(uri);
+    const char* query = evhttp_uri_get_query(uri_parsed);
+    std::optional<std::string> result;
+
+    if (query) {
+        // Parse the query string into a key-value queue and iterate over it
+        struct evkeyvalq params_q;
+        evhttp_parse_query_str(query, &params_q);
+
+        struct evkeyval* param;
+        for (param = params_q.tqh_first; param; param = param->next.tqe_next) {
```

Comment:
I'd suggest a couple of changes:

- put the `param` declaration in the for loop init-statement
- explicitly compare `param` with `nullptr`:

```suggestion
        for (evkeyval* param{params_q.tqh_first}; param != nullptr; param = param->next.tqe_next) {
```

Reply:
Done - although as per my other [comment](https://github.com/bitcoin/bitcoin/pull/24098#discussion_r816865897) I've kept the `struct` prefix for now to stick with convention.